crates/turborepo-lib/src/task_hash.rs

```diff
@@ -283,80 +286,51 @@ impl<'a> TaskHasher<'a> {
             .hashes
             .get(task_id)
             .ok_or_else(|| Error::MissingPackageFileHash(task_id.to_string()))?;
-        let mut explicit_env_var_map = EnvironmentVariableMap::default();
-        let mut all_env_var_map = EnvironmentVariableMap::default();
-        let mut matching_env_var_map = EnvironmentVariableMap::default();
-
-        let framework_slug = if do_framework_inference {
-            // See if we infer a framework
-            if let Some(framework) = infer_framework(workspace, is_monorepo) {
+        // See if we infer a framework
+        let framework = do_framework_inference
```

Comment:
Previously we had `if do_framework_inference { if let Some(framework) = infer_framework() { A } else { B } } else { C }` where `B` and `C` were the same statements. We now merge these two outcomes to a single branch.